### ISSUE TITLE
Browser assets cache reset

### ISSUE THREAD

**Issue by fullofcaffeine** [ashes999/butterfly#11]
Hi,

I'm experimenting with butterfly and I'm wondering how you'd deal with cleaning the browser cache when you push a new version to be served by the web server. I know there are headers that you can use and meta tags, and there's also the option of adding a unique hash to as a suffix to the assets files (middleman and jekyll have this out-of-the-box). 

How do you deal with this currently for your butterfly-genereted sites?

**Comment by ashes999** [comment]
Hi,

To be honest, I've never thought about or dealt with the issue. Refreshing a few times (or `CTRL`+`SHIFT`+`R` in Chrome) cleans the cache and shows me the updates.

Obviously, you can't tell this to consumers of your site. Unless you're publishing multiple new posts a day, I don't think this will really affect them.

One option I've seen used in the past is to append a random query string parameter to each request, eg. `?r=123281` which is generated automatically on the server-side. I don't know if that's an option worth investigating.

Headers and meta-tags might be the path of least resistance. I don't use Butterfly any more, but feel free to fork it and/or open a PR if that fits your needs.

**Comment by fullofcaffeine** [comment]
Thanks for the insights. I'll probably implement some kind of asset hash. 

> I don't use Butterfly any more, but feel free to fork it and/or open a PR if that fits your needs.
Oh, are you using something else for your static sites/blogs?

Cheers!

**Comment by ashes999** [comment]
> Oh, are you using something else for your static sites/blogs?

No, I simply haven't felt a need to blog in a long time. And my main site uses something custom (not Butterfly), because it has different requirements.

**Comment by fullofcaffeine** [comment]
Oh ok.

There's some appeal to using a site generator and Haxe is a nice option
because some targets are much faster than Ruby. Ruby does have
excelent generators like Jekyll and Middleman but it eventually becomes slow
as hell for bigger sites.

I think that for technical people, site generators are better than resident apps
(like Wordpress) because we're used to formats like Markdown and the
overhead of serving static files is much lower than hosting the actual app in memory
with caching etc. That's why I think Butterfly has some potential :)
